prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c

```diff
@@ -75,35 +75,97 @@ fi_ibv_rdm_start_connection(struct fi_ibv_rdm_ep *ep,
 	return FI_SUCCESS;
 }
 
-ssize_t fi_ibv_rdm_start_disconnection(struct fi_ibv_rdm_conn *conn)
+static inline ssize_t
+fi_ibv_rdm_disconnect_request(struct fi_ibv_rdm_ep *ep,
+			      struct fi_ibv_rdm_conn *conn)
+{
+	struct fi_ibv_rdm_buf *sbuf = NULL;
+	ssize_t ret = FI_SUCCESS;
+
+	if (!ep->is_closing) {
+		pthread_mutex_lock(&ep->cm_lock);
+	}
+	/* The latest control message MUST be in order */
+	while (conn->postponed_entry) {
+		fi_ibv_rdm_tagged_poll(ep);
+	}
+
+	/*
+	 * Wait untill resources will be available.
+	 * (Remote side should process all recvs then release local buffer)
+	 */
+	while ( !(sbuf = fi_ibv_rdm_prepare_send_resources(conn, ep)) ) {
+		if (conn->state != FI_VERBS_CONN_ESTABLISHED) {
+			goto out;
+		}
+	};
```

Comment:
I'll add check for QP state by ibv_query_qp. That should help to handle remote crash correctly